fuzz/fuzz_targets/node_eq.rs

```diff
@@ -1,12 +1,17 @@
 use clvmr::{Allocator, NodePtr, SExp};
+use std::collections::HashSet;
 
 /// compare two CLVM trees. Returns true if they are identical, false otherwise
 pub fn node_eq(allocator: &Allocator, lhs: NodePtr, rhs: NodePtr) -> bool {
     let mut stack = vec![(lhs, rhs)];
+    let mut visited = HashSet::<NodePtr>::new();
 
     while let Some((l, r)) = stack.pop() {
         match (allocator.sexp(l), allocator.sexp(r)) {
             (SExp::Pair(ll, lr), SExp::Pair(rl, rr)) => {
+                if !visited.insert(l) {
```

Comment:
avoiding comparing the same sub-trees multiple times is critical to make this function perform under a fuzzer